crates/e2e/src/setup/solver.rs

```diff
@@ -0,0 +1,245 @@
+    model::{
+        order::{BuyTokenDestination, OrderData, OrderKind, SellTokenSource},
+        signature::EcdsaSigningScheme,
+        DomainSeparator,
+    },
+    solvers_dto::{
+        auction::Auction,
+        solution::{Asset, Kind, Solution, Solutions, Trade},
+    },
+    std::{future::Future, net::SocketAddr, sync::Arc},
+    tokio::{
+        signal::{unix, unix::SignalKind},
+        sync::oneshot,
+    },
+    tracing::Instrument,
+    warp::hyper,
+    web3::signing::SecretKeyRef,
+};
+
+pub struct Mock {
+    solution: Solution,
+}
+
+impl Mock {
+    pub fn new(solution: Solution) -> Self {
+        Self { solution }
+    }
+
+    /// Returns the specified solution with the same order UID as the order UID
+    /// in the auction.
+    pub async fn solve(&self, auction: Auction) -> Solutions {
+        let mut solution = self.solution.clone();
+        // Return the same order UID as the order UID in the auction
+        solution
+            .trades
+            .iter_mut()
+            .filter_map(|trade| match trade {
+                Trade::Fulfillment(fulfillment) => Some(fulfillment),
+                Trade::Jit(_) => None,
+            })
+            .zip(auction.orders.iter())
+            .for_each(|(fulfillment, order)| fulfillment.order = order.uid);
+
+        Solutions {
+            solutions: vec![solution],
+        }
+    }
+}
+
+#[derive(Debug)]
+pub struct Config {
+    pub log: String,
+    pub addr: SocketAddr,
+    pub solution: Solution,
+}
+
+pub async fn run_mock(config: Config, bind: Option<oneshot::Sender<SocketAddr>>) {
+    observe::tracing::initialize_reentrant(&config.log);
+    tracing::info!("running mock solver engine with {config:#?}");
+
+    let solver = Mock::new(config.solution);
+
+    Api {
+        addr: config.addr,
+        solver,
+    }
+    .serve(bind, shutdown_signal())
+    .await
+    .unwrap();
+}
+
+pub async fn solve(
+    state: axum::extract::State<Arc<Mock>>,
+    Json(auction): Json<Auction>,
+) -> (axum::http::StatusCode, Json<Response<Solutions>>) {
+    let handle_request = async {
+        let auction_id = auction.id.unwrap_or_default();
+        let solutions = state
+            .solve(auction)
+            .instrument(tracing::info_span!("auction", id = %auction_id))
+            .await;
+
+        tracing::trace!(?auction_id, ?solutions);
+
+        (axum::http::StatusCode::OK, Json(Response::Ok(solutions)))
+    };
+
+    handle_request
+        .instrument(tracing::info_span!("/solve"))
+        .await
+}
+
+#[derive(Debug, Serialize)]
+#[serde(untagged)]
+pub enum Response<T> {
+    Ok(T),
+    Err(Error),
+}
+
+#[derive(Debug, Serialize)]
+pub struct Error {
+    pub message: &'static str,
+}
+
+impl From<&'static str> for Error {
+    fn from(message: &'static str) -> Self {
+        Self { message }
+    }
+}
+
+const REQUEST_BODY_LIMIT: usize = 10 * 1024 * 1024;
+
+pub struct Api {
```

Comment:
This file is cramped with many things. I would/will split it in a better way. Just did it like that in a rush to have it ready.